### ISSUE TITLE
MSYS2: unable to read the character under cursor

### ISSUE THREAD

**Issue by Andre9642** [nvaccess/nvda#11660]
### Steps to reproduce:
1. Open MSYS2.
2. Open a text file in an editor (like nano, emacs, vim).
3. Navigate with arrow keys.

### Actual behavior:
Whatever the character under cursor, "space" is spoken and '⠀' is displayed in braille.

### Expected behavior:
The appropriate character under cursor should be spoken/displayed.

#### NVDA installed/portable/running from source:
Any

#### NVDA version:
alpha-20935,c4de9f99

#### Windows version:
2004

#### Name and version of other software in use when reproducing the issue:
MSYS2 + nano/emacs/vim

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

#### If addons are disabled, is your problem still occuring?
Yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

**Comment by lukaszgo1** [comment]
What character is announced when reading current character at the review cursor position in this case? Could you try with NVDA 2019.2.1?

**Comment by Andre9642** [comment]
Also, I encounter other strange behaviors:

- If the cursor is under the first character of a line, the line is repeated (once including the first character, once not).
- If I move to the previous line using leftarrow key (from the first character of the current line), the next line is announced.

Here [a small demo](http://dl.andreabc.net/MSYS2.mp3).

> What character is announced when reading current character at the review cursor position in this case?

"Space".

> Could you try with NVDA 2019.2.1?

Same behavior.

**Comment by JulienCochuyt** [comment]
@Andre9642,
Did this behavior appear as of an update of MSYS2 or NVDA?
What version of MSYS2 are you using?
Unlike in #11842, you do not report here issues at the command prompt. What terminal and what shell are you using with MSYS2?

cc @codeofdusk 
cc @jeanbrazcosta

**Comment by Andre9642** [comment]
> Unlike in #11842, you do not report here issues at the command prompt. What terminal and what shell are you using with MSYS2?

Interesting. I've just retested it and the issue occurs at the command prompt as well. It seems to me that was not the case when I reported the issue...

> What version of MSYS2 are you using?

“MSYS2 is basically just a collection of packages installed on your computer, managed by a package manager, which is itself a package. There is no overall version number.” — [source](https://stackoverflow.com/questions/55559604/how-can-i-find-out-the-version-of-msys2-from-its-bash-shell#answer-55562058)

Currently I'm running on mintty 3.4.3 and bash 4.4.023-2.

I'll make more tests later.

**Comment by jeanbrazcosta** [comment]
Yes, all the tests I did when reporting the bug was with the MSYS2 git bach
terminal

Em sáb., 12 de dez. de 2020 às 06:58, Julien Cochuyt <
notifications@github.com> escreveu:

**Comment by JulienCochuyt** [comment]
Tested as far back as NVDA 2016.3: Same behavior.

@Andre9642, could you please try to downgrade to an earlier version of MinTTY in order to narrow-down when the issue appeared?

It might also be interesting to know if Cygwin users experience the same behavior with MinTTY or one of the other terminals it provides

**Comment by Adriani90** [comment]
@jeanbrazcosta, @aaclause are you still having this issue with NvDA 2024.3 beta in last MSYS?

**Comment by jeanbrazcosta** [comment]
@Adriani90  It was resolved for me too
